src/gamma/mod.rs

```diff
@@ -68,7 +63,9 @@ pub fn is_method_available(method_name: &str) -> bool {
 pub fn init_gamma_method(method_name: Option<&str>) -> Result<Box<GammaMethod>> {
     match method_name {
         Some(m) => {
-            SUPPORTED_GAMMA_METHODS[m]()
+            SUPPORTED_GAMMA_METHODS.get(m)
```

Comment:
I didn't want this - I would prefer that `main.rs` checks that the gamma method is available and errors out before calling `init_gamma_method()`. I wanted `init_gamma_method(Some(..))` to always succeed.